python/triqs_dft_tools/converters/plovasp/proj_shell.py

```diff
@@ -322,15 +316,26 @@ def density_matrix(self, el_struct, site_diag=True, spin_diag=True):
         occnums = el_struct.ferw
         ib1 = self.ib_min
         ib2 = self.ib_max + 1
+        # VASP.6.
+        count_nan = 0
+        print("Site diag : {}".format(site_diag))
         if site_diag:
             for isp in range(ns):
                 for ik, weight, occ in zip(it.count(), kweights, occnums[isp, :, :]):
                     for io in range(nion):
                         proj_k = self.proj_win[io, isp, ik, ...]
-                        occ_mats[isp, io, :, :] += np.dot(proj_k * occ[ib1:ib2],
-                                                     proj_k.conj().T).real * weight
-                        overlaps[isp, io, :, :] += np.dot(proj_k,
-                                                     proj_k.conj().T).real * weight
+                        # VASP.6.
+                        array_sum = np.sum(proj_k)
+                        if np.isnan(array_sum) == True: 
+                            count_nan += 1
+                        if self.nc_flag == True:
+                            occ_mats[isp, io, :, :] += 0.5 * np.dot(proj_k * occ[ib1:ib2], proj_k.T.conj()).real * weight
+                            overlaps[isp, io, :, :] += np.dot(proj_k, proj_k.T.conj()).real * weight
+                        else:
+                            occ_mats[isp, io, :, :] += np.dot(proj_k * occ[ib1:ib2], proj_k.T.conj()).real * weight
+                            overlaps[isp, io, :, :] += np.dot(proj_k, proj_k.T.conj()).real * weight
+            assert count_nan == 0, "!!! WARNING !!!: There are %s  NaN in your Projectors"%(count_nan)
```

Comment:
Ah I guess this is now the relevant check? Do we also need to check if there are negative / imaginary entries? If so, please remove the commented lines above completely.